submissions/SerhiiYakovenko/memory-pair-game/script.js

```diff
@@ -0,0 +1,128 @@
+const cards = [
+  {
+    name: "alt",
+    img: "img/alt.png",
+  },
+  {
+    name: "hanako",
+    img: "img/hanako.png",
+  },
+  {
+    name: "kerry_eurodyne",
+    img: "img/kerry_eurodyne.png",
+  },
+  {
+    name: "saburo",
+    img: "img/saburo.png",
+  },
+  {
+    name: "silverhand",
+    img: "img/silverhand.png",
+  },
+  {
+    name: "smasher",
+    img: "img/smasher.png",
+  },
+  {
+    name: "v-female",
+    img: "img/v-female.png",
+  },
+  {
+    name: "v-male",
+    img: "img/v-male.png",
+  },
+];
+
+const div = document.createElement("div");
+div.id = "board";
+document.body.appendChild(div);
+
+let mixedCards = cards.concat(cards).sort(function () {
+  return 0.5 - Math.random();
+});
+
+let firstFlip = "";
+let secondFlip = "";
+let turn = 0;
+let previousCard = null;
+
+const board = document.getElementById("board");
+const boardGrid = document.createElement("section");
+boardGrid.setAttribute("class", "boardGrid");
+board.appendChild(boardGrid);
+
+mixedCards.forEach(function (item) {
+  const card = document.createElement("div");
+  const front = document.createElement("div");
+  const back = document.createElement("div");
+
+  card.classList.add("card");
+  card.dataset.name = item.name;
+  front.classList.add("front");
+  back.classList.add("back");
+  back.style.backgroundImage = `url('${item.img}')`;
+
+  boardGrid.appendChild(card);
+  card.appendChild(front);
+  card.appendChild(back);
+});
+
+let match = function match() {
+  let selected = document.querySelectorAll(".selected");
+  selected.forEach(function (card) {
+    card.classList.add("match");
+  });
+  if (document.getElementsByClassName("card match").length == 16) {
+    setTimeout(alert("Congrats! You won!"), 3000);
+  }
+};
+
+let reset = function reset() {
+  firstFlip = "";
+  secondFlip = "";
+  turn = 0;
+  previousCard = null;
+
+  let selected = document.querySelectorAll(".selected");
+  selected.forEach(function (card) {
+    card.classList.remove("selected");
+  });
+};
+
+boardGrid.addEventListener("click", function (event) {
+  let target = event.target;
+  if (
+    target.nodeName === "SECTION" ||
+    target === previousCard ||
+    target.parentNode.classList.contains("selected") ||
+    target.parentNode.classList.contains("match")
+  ) {
+    return;
+  }
+  if (turn < 2) {
```

Comment:
2 
[Magic number](https://en.wikipedia.org/wiki/Magic_number_(programming))

Reply:
hm, not sure that 2 is the magic number, as it's the number of turns available to the user to swap 2 cards, as this is game rules. 
I can create some kind of const like:
`const MAX_TURNS = 2`, but as for me, it's redundant. 
wdyt?

Reply:
> as this is game rules.
What if game rules will change ? if this number become 8 (for example)
Moving such things to a variable makes your code more readable for other people 
```
if (turn < MAX_TURNS) // is more obvious 
```
```
if (turn < 8) // why 4 ?
```

Reply:
Created new const called `MAX_FLIPS`